queries/kotlin/highlights.scm

```diff
@@ -0,0 +1,162 @@
+;;; Kotlin 
+;; Issues
+
+; mapOf with type definition AND multiple entries are not highligted correctly. Issue in parser. Eg.
+; a = mapOf<String, String>(AA.value to AA.value, AA.value to AA.value)
+
+
+(function_modifier) @type
+
+;; String
+
+(line_string_literal) @string
+(multi_line_string_literal) @string
+(line_string_literal (interpolated_identifier)) @string
+(line_string_literal (interpolated_expression (line_string_literal))) @string
+
+;; Constants
+; Assume all-caps names are constants
+((simple_identifier) @constant
+ (#vim-match? @constant "^[A-Z][A-Z_]+"))
+
+
+;; Variables
+(class_parameter (simple_identifier) @variable.builtin)
+
+; Class level
+(property_declaration (variable_declaration (simple_identifier) @variable.builtin))
```

Comment:
Can you explain what simple_identifier is that it is `@variable.builtin`? `@variable.builtin` is usually for keyword like variables like `this` or `self`

Reply:
sorry for abusing some of the highlight categories. for the first version I just wanted to highlight the different categories in different colors. I'll go through these once more. this query will hit `variableA` in things like `private val variableA: Int = 2`. maybe `@field` is the correct one to use here?

I plan to move this to support local scoping if I find out how that works

Reply:
yes, `@field` or `@property`